src/Files.App/Utils/Storage/Collection/GroupingHelper.cs

```diff
@@ -120,7 +120,7 @@ public static (Action<GroupedCollection<ListedItem>>, Action<GroupedCollection<L
 					var model = x.Model;
 					model.ShowCountTextBelow = true;
 					var parentPath = PathNormalization.GetParentDir(first.ItemPath.TrimPath());
-					model.Text = System.IO.Path.GetFileName(parentPath);
+					model.Text = IsUNCPath(parentPath) ? GetUNCFileName(parentPath) : System.IO.Path.GetFileName(parentPath);
```

Comment:
This is not just a UNC problem, but a problem when the parent is not a folder. What if the library location is the route of a drive?

Reply:
Yeah I guess so. How do I go about it?

Reply:
You can check `System.IO.Path.GetFileName()` first. If `System.IO.Path.GetFileName()` is null or empty, then you can try `GetUNCFileName()`. The method name should be changed though.

Reply:
For all the System.IO.Path.GetFileName() in the project?

Reply:
Just here is fine.

Reply:
Okay

Reply:
Resolved Please check